logger.go

```diff
@@ -0,0 +1,50 @@
+package llog
+
+import (
+	"log"
+	"strings"
+)
+
+type llogWriter struct {
+	fn      LogFunc
+	kv      KV
+	filters []func(string) (string, error)
+}
+
+// Write implements the io.Writer interface
+func (lw *llogWriter) Write(b []byte) (int, error) {
+	msg := strings.TrimSpace(string(b))
+	for _, fn := range lw.filters {
+		m, err := fn(msg)
+		if err != nil {
+			return 0, err
+		}
+		// ignore if there's no message left
+		if m == "" {
+			// pretend we still wrote everything
+			return len(b), nil
+		}
+		msg = m
+	}
+	// TODO: what should we do with multi-line messages
+	lw.fn(string(msg), lw.kv)
```

Comment:
Isn't `msg` guaranteed to be a string here?

Reply:
Good catch, I guess I used to not have the `strings.TrimSpace`